src/ui/hash.js

```diff
@@ -15,9 +15,12 @@ import type Map from './map';
 class Hash {
     _map: Map;
     _updateHash: () => ?TimeoutID;
+    _hashName: ?string;
 
-    constructor() {
+    constructor(hashName: ?string) {
+        this._hashName = hashName;
```

Comment:
We should probably do some validation here. If I set `hash: 'hello world'` or `hash: 'hello\nworld'`, the hash is appended twice so it makes sense to ensure the hash is what we expect.

Reply:
What do you recommend, display a warning/error or sanitize the hashName?